### DISCUSSION TITLE
equals, hashcode는 일단 재정의 해야한다 vs 필요할때 재정의 하면된다

### DISCUSSION THREAD

**Discussion by seong-wooo** [woowa-Effective-Java-Study/effective-java-study#9] [General]
의견을 나누어 보아요

**Comment by OzRagwort** [comment]
일반적으로는 같이 개발을 하다보면 (구현코드를) 어떤 클라이언트가 쓸 지 모르기 때문에 동일성 규칙이 정해져잇다면 재정의 해둬야 좋다. 

아그니 원칙? -> 필요할때만 원칙이 존재하긴 하지만 앞으로가는걸 만들면 ㄴ뒤로가는것도 만들고 추가하는 기능을 만들면 삭제하는 기능(대칭되는 기능 등)을 미리 만들어 두는 편이다.

**Comment by yxxnghwan** [comment]
yagni
